RealmJS.podspec

```diff
@@ -41,67 +41,29 @@ Pod::Spec.new do |s|
-                               'CURRENT_PROJECT_VERSION' => s.version,
-                               'VERSIONING_SYSTEM' => 'apple-generic',
-                               # Disable C++17 features for unsupported platforms
-                               'OTHER_CPLUSPLUSFLAGS[arch=armv7]' => '-fno-aligned-new',
-                               # Header search paths are prefixes to the path specified in #include macros
-                               'HEADER_SEARCH_PATHS' => [
-                                 '"$(PODS_TARGET_SRCROOT)/src/"',
-                                 '"$(PODS_TARGET_SRCROOT)/src/jsc/"',
-                                 '"$(PODS_TARGET_SRCROOT)/src/object-store/src/"',
-                                 '"$(PODS_TARGET_SRCROOT)/src/object-store/external/json/"',
-                                 '"$(PODS_TARGET_SRCROOT)/vendor/"',
-                                 '"$(PODS_TARGET_SRCROOT)/vendor/realm-ios/include/"',
-                                 '"$(PODS_TARGET_SRCROOT)/vendor/realm-ios/include/realm/"',
-                                 '"$(PODS_TARGET_SRCROOT)/react-native/ios/RealmReact/"',
-                                 '"$(PODS_ROOT)/Headers/Public/React-Core/"'
-                                 # "'#{app_path}/ios/Pods/Headers/Public/React-Core'" # Use this line instead of 👆 while linting
-                               ].join(' ')
-                             }
+
+  s.pod_target_xcconfig    = {
+                                # Setting up clang
+                                'CLANG_CXX_LANGUAGE_STANDARD' => 'c++17',
+                                'CLANG_CXX_LIBRARY' => 'libc++',
+                                # Setting the current project version and versioning system to get a symbol for analytics
+                                'CURRENT_PROJECT_VERSION' => s.version,
+                                'VERSIONING_SYSTEM' => 'apple-generic',
+                                # Header search paths are prefixes to the path specified in #include macros
+                                'HEADER_SEARCH_PATHS' => [
+                                  '"$(PODS_TARGET_SRCROOT)/react-native/ios/RealmReact/"',
+                                  '"$(PODS_ROOT)/Headers/Public/React-Core/"'
+                                  #"'#{app_path}/ios/Pods/Headers/Public/React-Core'" # Use this line instead of 👆 while linting
+                                ].join(' ')
+                              }
 
   # TODO: Consider providing an option to build with the -dbg binaries instead
-  s.ios.vendored_libraries = 'vendor/realm-ios/librealm-sync-ios.a', 'vendor/realm-ios/librealm-parser-ios.a'
-  # s.watchos.vendored_libraries = 'vendor/realm-ios/librealm-sync-watchos.a', 'vendor/realm-ios/librealm-parser-watchos.a'
-  # s.tvos.vendored_libraries = 'vendor/realm-ios/librealm-sync-tvos.a', 'vendor/realm-ios/librealm-parser-tvos.a'
+  s.ios.vendored_frameworks = 'react-native/ios/realm-js-ios.xcframework'
```

Comment:
Can we easily get support for Catalyst, tvOS, and watchOS?

Reply:
Core has toolchains to easily build for all those, and the xcframework can fit the slices, too. I'm not sure how to plug them in to React Native, though.

Reply:
I suppose we need to specify alternative [multi-platform values](https://guides.cocoapods.org/syntax/podspec.html#group_multi_platform_support) for the `vendored_frameworks` and build those as well?

```podspec
s.macos.vendored_frameworks = 'react-native/ios/realm-js-macos.xcframework'
s.tvos.vendored_frameworks = 'react-native/ios/realm-js-tvos.xcframework'
// ...
```